### ISSUE TITLE
[Swift ] CCSizeTypeNormalized and CCPositionTypeNormalzied error

### ISSUE THREAD

**Issue by collidernyc** [spritebuilder/SpriteBuilder#1346]
http://forum.cocos2d-swift.org/t/ccsizetypenormalized-and-ccpositiontypenormalzied-error-in-swift/16638?u=collidernyc

When setting up a sprite, I'm trying to do the following:

var turret = CCSprite(CCSpriteFrameCache.sharedSpriteFrameCache().spriteFrameByName("turret.png"))
turret.position = ccp(0.5,0.5)

//this line causes an error:
turret.positionType = CCPositionTypeNormalized

self.addChild(turret)
The error comes up as:

Undefined symbols for architecture armv7:
  "_CCSizeTypeNormalized", referenced from:
      __TFC12TestingSwift9MainScenecfMS0_FT_S0_ in MainScene.o
ld: symbol(s) not found for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation)

A very similar thing happens with CCSizeTypeNormalized

**Comment by Ben-G** [comment]
Apparently the Swift compiler treats static constants in headers differently then the Objective-C compiler:
http://stackoverflow.com/questions/28577147/not-allowing-to-use-constant-defined-in-objective-c-header-file-in-swift-class

Potentially the Swift compiler changes in future to support this. We could fix this by declaring all of the constants as extern and adding a .m file for their definition.

**Comment by sana20** [comment]
According to @Ben-G  comment this issue is ready to be closed.

**Comment by collidernyc** [comment]
As of SB 1.4.7 and Xcode Version 6.3 (6D543q) it is still an issue. Has there been a change in the SB beta to fix this?

**Comment by Ben-G** [comment]
@sana20 No. The issue is not closed. The constants will need be declared using `extern` in the .h file, and be defined in the .m file.

**Comment by sana20** [comment]
@Ben-G  Thank you for clarifying this issue. We treat it as the open issue.
